src/MeshlessFV/MeshlessFVTree.cpp

```diff
@@ -491,6 +491,8 @@ void MeshlessFVTree<ndim,ParticleType,TreeCell>::UpdateGradientMatrices
         // Make local copy of hmask for active particle
         hmask = mfv->types[activepart[j].ptype].hmask;
 
+        activepart[j].levelneib = 0;
+
```

Comment:
Yes you're right, I made confusion between this and the zero-ing of the levelneib array before the beginning of the loop over cells. Btw, I realise that that is done only up to Nhydro; surely this is wrong and should be up to Ntot...